### ISSUE TITLE
VPN is not connecting in IOS. Remains on the state: "Connecting".

### ISSUE THREAD

**Issue by SyedHaadi** [ccnnde/react-native-simple-openvpn#60]
**Bug description:**

**Environment:**
 - OS: IOS
 - OS version: Any version
 - react-native version: 0.71.1
 - react-native-simple-openvpn version: latest version

**Comment by ccnnde** [comment]
Have you ever tried the Android platform? And, the official OpenVPN app, can it connect normally?

**Comment by SyedHaadi** [comment]
> Have you ever tried the Android platform? And, the official OpenVPN app, can it connect normally?

Yes, With that .opvn file, it connects normally on both Android and OpenVPN App.

**Comment by SyedHaadi** [comment]
![Img1](https://github.com/ccnnde/react-native-simple-openvpn/assets/88819339/3c4c7fd5-f63f-42e1-b137-6b603d96bde0)
![Img2](https://github.com/ccnnde/react-native-simple-openvpn/assets/88819339/88302afd-98c1-4595-8713-616514036e62)

You can see the project structure and the capabilities I added. And also the OpenVPN Adapter version in this.

**Comment by SyedHaadi** [comment]
If you can help me with this. It really will be appreciated.

**Comment by SyedHaadi** [comment]
I tried this in two different ways. 
Firstly I tried the method mentioned in the IOS Guide , Create the Cartfile file in the ios/ directory, add github "ss-abramchuk/OpenVPNAdapter", and run carthage update and Drag the OpenVPNAdapter.framework.
Second method which I tried is, I go to the Add Package in xcode and search for the package by name and install it .
But unfortunately both methods didn't work.

**Comment by SyedHaadi** [comment]
> How do you install OpenVPNAdapter? I didn't see Package Dependencies in XCode for my project.

I tried this in two different ways.
Firstly I tried the method mentioned in the IOS Guide, Create the Cartfile file in the ios/ directory, add github "ss-abramchuk/OpenVPNAdapter", and run carthage update and Drag the OpenVPNAdapter.framework.
Second method which I tried is, I go to the Add Package in xcode and search for the package by name and install it .
But unfortunately both methods didn't work.

**Comment by ccnnde** [comment]
I think this issue is not easy to handle. I'm sorry.

**Comment by SyedHaadi** [comment]
> I think this issue is not easy to handle. I'm sorry.

If you want I will share the zip of my project with you If you can check and help me. It's essential for me please do something if you can.

**Comment by KevinCHAILLY** [comment]
Did you try to run the ovpn file inside the example app ?

**Comment by SyedHaadi** [comment]
> Did you try to run the ovpn file inside the example app ?

Do you mean inside the main directory of the app?

**Comment by omarsaade** [comment]
This is my code   ```
import React, {useEffect, useState} from 'react';
import {Platform, View, Text, Button} from 'react-native';
import RNSimpleOpenvpn, {
  addVpnStateListener,
  removeVpnStateListener,
} from 'react-native-simple-openvpn';

const isIPhone = Platform.OS === 'ios';

const App = () => {
  const [vpnStatus, setVpnStatus] = useState('Disconnected');

  useEffect(() => {
    async function observeVpn() {
      if (isIPhone) {
        await RNSimpleOpenvpn.observeState();
      }

      addVpnStateListener(e => {
        // Update VPN status
        setVpnStatus(e.state);
      });
    }

    observeVpn();

    return async () => {
      if (isIPhone) {
        await RNSimpleOpenvpn.stopObserveState();
      }

      removeVpnStateListener();
    };
  }, []);

  async function startOvpn() {
    try {
      await RNSimpleOpenvpn.connect({
        remoteAddress: '192.168.1.1 3000',
        ovpnFileName: 'client',
        assetsPath: 'ovpn/',
        providerBundleIdentifier: 'com.example.RNSimpleOvpnTest.NEOpenVPN',
        localizedDescription: 'RNSimpleOvpn',
      });
    } catch (error) {
      console.error('Error connecting to VPN:', error);
    }
  }

  async function stopOvpn() {
    try {
      await RNSimpleOpenvpn.disconnect();
    } catch (error) {
      console.error('Error disconnecting from VPN:', error);
    }
  }

  return (
    <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
      <Text>VPN Status: {vpnStatus}</Text>
      <Button title="Connect" onPress={startOvpn} />
      <Button title="Disconnect" onPress={stopOvpn} />
    </View>
  );
};

export default App;

```
And  this is the error  it returns. Note that I'm not an expert in VPN or these services.  Can someone help me understand how to continue and how to solve this error "Error connecting to VPN: [Error: Prepare VPN failed]"?  What am I missing?

**Comment by Vishujain1** [comment]
@SyedHaadi is your android app working ?

**Comment by Luthamm** [comment]
Any updates on this? Running into the same issue now.
